uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/grpc/GRPCControlServer.java

```diff
@@ -279,6 +280,7 @@ public void publishMqtt(MqttPublishRequest request, StreamObserver<MqttPublishRe
                     .retain(isRetain)
                     .topic(topic)
                     .payload(message.getPayload().toByteArray())
+                    .userProperties(message.getProperties().getUserPropertiesMap())
```

Comment:
Please check user properties only in case when it actually present in the request.